oak/server/rust/oak_loader/src/lib.rs

```diff
@@ -0,0 +1,229 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Helper functions to parse input arguments and create an instance of
+//! `RuntimeConfiguration` to feed into the oak_loader.
+
+use anyhow::{anyhow, Context};
+use core::str::FromStr;
+use log::debug;
+use oak_abi::proto::oak::application::{ApplicationConfiguration, ConfigMap};
+use oak_runtime::{
+    auth::oidc_utils::{parse_client_info_json, ClientInfo},
+    config::load_certificate,
+};
+use prost::Message;
+use std::{
+    collections::HashMap,
+    fs::{read, read_to_string},
+};
+use structopt::StructOpt;
+use tonic::transport::Identity;
+
+/// Command line options for the Oak loader.
+#[derive(StructOpt, Clone, Debug)]
+#[structopt(about = "Oak Loader")]
+pub struct Opt {
```

Comment:
I am not sure what the reason is for moving this to separate library crate rather than a submodule.

If the plan is to reuse it from other binaries, it might be better to not include the command-line arguments. Command-line flags used in libraries often lead to conflicts or confusion, especially if multiple libraries try to declare their own flags.

If there are no plans for reuse, it might be better to just move it to a submodule rather that a separate library crate.

Reply:
I am also unsure why this was all moved to a lib.rs file, what's the intention behind it @rbehjati ?

Reply:
Yes, I think a submodule is better. Perhaps also change the title of the PR to remain consistent?

Reply:
Good point. Thanks.